File: pickphotoview/src/main/java/com/werb/pickphotoview/PickPhotoView.java
package com.werb.pickphotoview;

import android.app.Activity;
import android.content.Intent;

import com.werb.pickphotoview.model.PickData;
import com.werb.pickphotoview.util.PickConfig;

/**
 * Created by wanbo on 2016/12/30.
 */

public class PickPhotoView {

    private PickData pickData;
    private Activity activity;

    private PickPhotoView(Builder builder){
        pickData = builder.pickData;
        activity = builder.activity;
    }

    private void start(){
        Intent intent = new Intent();
        intent.setClass(activity,PickPhotoActivity.class);
        intent.putExtra(PickConfig.INTENT_PICK_DATA,pickData);
        activity.startActivityForResult(intent,PickConfig.PICK_PHOTO_DATA);
    }

    public static class Builder{

        private PickData pickData;
        private Activity activity;

        public Builder(Activity a) {
            pickData = new PickData();
            activity = a;
        }

        public Builder setPickPhotoSize(int photoSize) {
            pickData.setPickPhotoSize(photoSize);
            return this;
        }

        public Builder setSpanCount(int spanCount) {
            pickData.setSpanCount(spanCount);
            return this;
        }

        public Builder setShowCamera(boolean showCamera) {
            pickData.setShowCamera(showCamera);
            return this;
        }

        public Builder setToolbarColor(String toolbarColor){
            pickData.setToolbarColor(toolbarColor);
            return this;
        }

        public Builder setStatusBarColor(String statusBarColor){
            pickData.setStatusBarColor(statusBarColor);
            return this;
        }

        public Builder setToolbarIconColor(String toolbarIconColor){
            pickData.setToolbarIconColor(toolbarIconColor);
            return this;
        }

        public Builder setLightStatusBar(boolean lightStatusBar){
            pickData.setLightStatusBar(lightStatusBar);
            return this;
        }

        private PickPhotoView create(){
            return new PickPhotoView(this);
        }

        public void start(){
            create().start();
        }

    }
}

File: pickphotoview/src/main/java/com/werb/pickphotoview/PickPhotoView.kt
package com.werb.pickphotoview

import android.app.Activity

import com.werb.pickphotoview.model.PickModel
import com.werb.pickphotoview.util.PickConfig

/**
 * Created by wanbo on 2016/12/30.
 */

class PickPhotoView private constructor(private val builder: Builder) {

    private fun start() {
        PickPhotoActivity.startActivity(builder.activity, PickConfig.PICK_PHOTO_DATA)
    }

    class Builder(val activity: Activity) {

        private val pickModel = PickModel()

        fun setPickPhotoSize(photoSize: Int): Builder {
            pickModel.pickPhotoSize = photoSize
            return this
        }

        fun setHasPhotoSize(hasSize: Int): Builder {
            pickModel.hasPhotoSize = hasSize
            return this
        }

        fun setAllPhotoSize(allSize: Int): Builder {
            pickModel.allPhotoSize = allSize
            return this
        }

        fun setSpanCount(spanCount: Int): Builder {
            pickModel.spanCount = spanCount
            return this
        }

        fun setShowCamera(showCamera: Boolean): Builder {
            pickModel.isShowCamera = showCamera
            return this
        }

        /** clickSelectable used with photoSize == 1  */
        fun setClickSelectable(clickSelectable: Boolean): Builder {
            pickModel.isClickSelectable = clickSelectable
            return this
        }

        fun setToolbarColor(toolbarColor: Int): Builder {
            pickModel.toolbarColor = toolbarColor
            return this
        }

        fun setStatusBarColor(statusBarColor: Int): Builder {
            pickModel.statusBarColor = statusBarColor
            return this
        }

        fun setToolbarTextColor(toolbarTextColor: Int): Builder {
            pickModel.toolbarTextColor = toolbarTextColor
            return this
        }

        fun setSelectIconColor(selectIconColor: Int): Builder {
            pickModel.selectIconColor = selectIconColor
            return this
        }

        fun setLightStatusBar(lightStatusBar: Boolean): Builder {
            pickModel.lightStatusBar = lightStatusBar
            return this
        }

        fun setShowGif(showGif: Boolean): Builder {
            pickModel.isShowGif = showGif
            return this
        }

//        fun setShowVideo(showVideo: Boolean): Builder {
//            pickModel.isShowVideo = showVideo
//            return this
//        }

        private fun create(): PickPhotoView {
            GlobalData.model = pickModel
            return PickPhotoView(this)
        }

        fun start() {
            create().start()
        }

    }
}
